Source/FullScreenPokemon.ts

```diff
@@ -2918,6 +2940,85 @@ module FullScreenPokemon {
          * @param direction   The direction border being changed.
          */
         setThingBordering(thing: IThing, other: IThing, direction: Direction): void {
+            if (thing.title === "Ledge") {
```

Comment:
Instead of hardcoding a reaction for ledges, a better way would add an `onBordering` hook (or some equivalent name) that Things could describe. Ledges could then declare it to be an FSP function named `setLedgeBordering` that would contain the logic.